src/main/kotlin/io/kotlintest/properties/Gen.kt

```diff
@@ -8,69 +8,34 @@ import java.util.*
 /** A shared random number generator. */
 private val RANDOM = Random()
 
-interface Gen<T> {
+interface Gen<out T> {
   fun generate(): T
 
   companion object {
-
-    fun choose(min: Int, max: Int): Gen<Int> = object : Gen<Int> {
-      override fun generate(): Int = RANDOM.nextInt((max.toLong() - min.toLong()).toInt()) + min
-    }
-
-    fun choose(min: Long, max: Long): Gen<Long> = object : Gen<Long> {
-      override fun generate(): Long {
-        var rand = (RANDOM.nextLong() % (max - min))
-        if (rand < 0) {
-          rand += max - min
-        }
-        return rand + min
-      }
-    }
-
-    fun <T> oneOf(vararg generators: Gen<T>): Gen<T> = object : Gen<T> {
-      override fun generate(): T = Gen.oneOf(generators.toList()).generate().generate()
-
-    }
-
-    fun <T> oneOf(values: List<T>): Gen<T> = object : Gen<T> {
-      override fun generate(): T = values[RANDOM.nextInt(values.size)]
-    }
-
-    fun string(): Gen<String> = object : Gen<String> {
-      override fun generate(): String = nextPrintableString(RANDOM.nextInt(100))
-    }
-
-    fun int() = object : Gen<Int> {
-      override fun generate(): Int = RANDOM.nextInt()
-    }
-
-    fun long() = object : Gen<Long> {
-      override fun generate(): Long = RANDOM.nextLong()
-    }
-
-    fun bool() = object : Gen<Boolean> {
-      override fun generate(): Boolean = RANDOM.nextBoolean()
-    }
-
-    fun double() = object : Gen<Double> {
-      override fun generate(): Double = RANDOM.nextDouble()
-    }
-
-    fun float() = object : Gen<Float> {
-      override fun generate(): Float = RANDOM.nextFloat()
-    }
-
-    fun <T> create(fn: () -> T): Gen<T> = object : Gen<T> {
+    inline fun <T> create(crossinline fn: () -> T): Gen<T> = object : Gen<T> {
```

Comment:
Inlining the `create` function (but the `fn` one is executed in a different context, make it crossinline)